src/Core/Components/DateTime/FluentDatePicker.razor.cs

```diff
@@ -76,7 +84,13 @@ protected Task OnSelectedDateAsync(DateTime? value)
 
     protected override bool TryParseValueFromString(string? value, out DateTime? result, [NotNullWhen(false)] out string? validationErrorMessage)
     {
+        if (Regex.IsMatch(value ?? "", @"^\d{4}$"))
```

Comment:
That's not correct: you must do that only when "View = Years".
And you can simplify the code using a `int.TryParse`.

Example
```csharp
if (View == CalendarViews.Years && int.TryParse(value, out int year))
{
    value = new DateTime(year, 1, 1).ToString(Culture.DateTimeFormat.ShortDatePattern);
}
```

Reply:
Not interested in it containging exactly 4 digits?